frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -307,11 +312,36 @@ This stylesheet is to be used only with FB2 and FB3 documents, which are not cla
             end
             return text
         end,
-        sub_item_table = obsoleted_table,
         checked_func = function()
             return obsoleted_css[self.css] ~= nil
-        end
+        end,
+        sub_item_table = obsoleted_table,
+        separator = true,
     })
+    if self.ui.document.is_txt then
+        table.insert(style_table, {
+            text_func = function()
+                return _("Auto-detect TXT files layout") .. (G_reader_settings:has("txt_preformatted") and "   ★" or "")
+            end,
+            checked_func = function()
+                return self.txt_preformatted == 0
+            end,
+            callback = function()
+                self.txt_preformatted = self.txt_preformatted == 1 and 0 or 1
+                self.ui.doc_settings:saveSetting("txt_preformatted", self.txt_preformatted)
+                -- setting txt_preformatted for the opened document causes segfault, hence reload
+                self.ui.rolling:showReloadConfirmBox()
```

Comment:
```
-- Calling document:setTxtPreFormatted() here could cause a segfault (there is something
-- really fishy about its handling, like bits of partial rerenderings happening while it is
-- disabled...). It's safer to just not notify crengine, and propose the user to reload the
-- document and restart from a sane state.
```

There's something really fishy I can't figure out... and won't have time, so fine with reloading the document and pretend there's no fish :)